Store.js

```diff
@@ -140,14 +150,109 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
 						collection.on("delete", this._itemRemoved.bind(this));
 						collection.on("refresh", this._refreshHandler.bind(this));
 					}
-					return this.processCollection(collection);
+				} else {
+					for (var i = 0; i < this.store.length; i++) {
+						// affect the callback to the observe function if the item is observable
+						Observable.observe(this.store[i], this.observeCallback.bind(this));
+					}
+					collection = processQueryResult.call(this, this.store.filter(this._isQueried, this));
+					// affect the callback to the observe function if the array is observable
+					this._observeResult = ObservableArray.observe(this.store, this.observeCallback.bind(this));
 				}
+				return this.processCollection(collection);
 			} else {
 				this.initItems([]);
 			}
 		},
 
 		/**
+		 * Synchronously deliver change records of the Observe function for the Array
+		 * Do NOT deliver change records of the Observe function for Objects
+		 */
+		deliver: dcl.superCall(function (sup) {
+			return function () {
+				sup.call();
+				if (this._observeResult !== null && this._observeResult !== undefined) {
+					this._observeResult.deliver();
+				}
+			};
+		}),
+
+		/**
+		 * Called to verify if the item respect the query conditions
+		 * @param item
+		 * @protected
+		 */
+		_isQueried: function (item) {
+			for (var prop in this.query) {
+				if (item[prop] !== this.query[prop]) {
+					return false;
+				}
+			}
+			return true;
+		},
+
+		/**
+		 * Called when a modification is done on the array or its items.
+		 * @param changeRecords - send by the Observe function
+		 */
+		observeCallback: function (changeRecords) {
+			for (var i = 0; i < changeRecords.length; i++) {
+				// array modified
+				if (Array.isArray(changeRecords[i].object) === true) {
```

Comment:
I did not think about that in my implementation. It may need some modifications.